test/Microsoft.AspNet.Mvc.Extensions.Test/Rendering/HtmlHelperTest.cs

```diff
@@ -54,5 +146,135 @@ public class HtmlHelperTest
             var entry = Assert.Single(result);
             Assert.Equal(expectedEntry, entry);
         }
+
+        [Theory]
+        [MemberData(nameof(EncodeDynamicTestData))]
+        public void EncodeDynamic_ReturnsExpectedString(object value, string expectedString)
```

Comment:
Adding lots of new tests. This method and `RawDynamic_ReturnsExpectedString()` are the ones that fail without `IHtmlHelper<TModel>` change.